gnc/sub8_perception/src/sub8_pcl/sub8_gateway.py

```diff
@@ -0,0 +1,202 @@
+__author__ = 'Erol Bahadiroglu'
+
+'''
+This script receives images from one of the submarine's cameras (stereo/left) and searches for a shape resembles a
+gateway. This involves applying the Harris Corner Detection tool (from OpenCV), constructing a set of vectors between
+the corner points, and finding overlapping pairs of perpendicular vectors.
+
+Output is sent with the topic 'gatway'. It is a JSON with the format [[top left corner], [dimensions]].
+'''
+
+import cv2
+import matplotlib.pyplot as plt
+import numpy as np
+import rospy
+from sensor_msgs.msg import Image as ROS_img
+from std_msgs.msg import String as ROS_string
+import cv_bridge
+import json
+
+camera = 'stereo/left'  # address of camera
+destination = 'gateway'  # topic where data is published
+#input_image = cv2.imread("images.png")  # testing image
+
+############### IMAGE PROCESSING ###############
+
+class Vector(object):
+    def __init__(self, unit_vector, magnitude, point1, point2):
+        self.unit = unit_vector
+        self.magnitude = magnitude
+        self.point_set = (point1, point2)
+
+    def __repr__(self):
+        return "unit vector: " + str(self.unit) + "; magnitude: " + str(self.magnitude) + "; " + str(self.point_set)
+
+    def equals(self, other):
+        if (self.magnitude == other.magnitude) and (set(self.unit) == set(other.unit)):

```

Comment:
Since magnitude is a floating point number, equality checks should be replaced with 

    if np.fabs(num_1 - num_2) < epsilon:
        print "hello"

Where epsilon is a small number, like 1e-5